modelon/impact/client/entities/case.py

```diff
@@ -127,6 +130,67 @@ def simulation_log_level(self, simulation_log_level: str):
         self._analysis['simulation_log_level'] = simulation_log_level
 
 
+@dataclass
+class CustomArtifact:
+    """CustomArtifact class"""
+
+    def __init__(
+        self,
+        workspace_id: str,
+        experiment_id: str,
+        case_id: str,
+        data: Dict[str, str],
+        exp_sal,
+    ):
+        self._workspace_id = workspace_id
+        self._exp_id = experiment_id
+        self._case_id = case_id
+        self._data = data
+        self._exp_sal = exp_sal
+
+    @property
+    def id(self):
+        """Id of the custom artifact."""
+        return self._data['id']
+
+    @property
+    def download_as(self):
+        """File name for the downloaded artifact."""
+        return self._data['downloadAs']
+
+    def download(self, path: Optional[str] = None):
+        """Downloads a custom artifact.
+        Returns the local path to the downloaded artifact.
+
+        Parameters:
+
+            path --
+                The local path to store the custom artifact. Default: None.
```

Comment:
It could be clarified that this local path should is a directory and the file is put in it.